src/ServiceControl/Infrastructure/Settings/Settings.cs

```diff
@@ -501,6 +504,27 @@ static string Subscope(string address)
             return $"{queue}.log@{machine}";
         }
 
+        int GetDataSpaceRemainingThreshold()
+        {
+            string message;
+            var threshold = SettingsReader<int>.Read("DataSpaceRemainingThreshold", DataSpaceRemainingThresholdDefault);
+            if (threshold < 0)
+            {
+                message = $"{nameof(DataSpaceRemainingThreshold)} is invalid, minimum value is 0.";
+                logger.Fatal(message);
+                throw new Exception(message);
```

Comment:
Pedantic comment, but is `Exception` the correct type?

Reply:
We could add a custom exception type. I was just following the pattern that is already in place. All of the other settings throw Exception if they are invalid. There's 17 other instances.